### PR TITLE
Improve changeset form UX

### LINKED ISSUE #8027: a8n: minor UI improvements
- [x] 1. Add field labels to form inputs, so they are visible when there is input text:
   image

- [x] 2. Confirm code hosts all use numeric changeset numbers, then make the changeset number input field a number field.

- [x] 3. Add helper text below the repo input field to make it clear what is expected as input: "This is the name of the repository in Sourcegraph in  ${window.location.protocol + window.location.host}/$REPOSITORY_NAME"

- [x] 4. Add some margin between the form inputs and the errors:
image

### PR DESCRIPTION
**Pull request by eseliger**
Closes #8027 

I also added some regex parsing to the pasted content, so it'll split bitbucket and github urls properly now and paste in what is expected. Lmk if you like that behavior or if it feels weird (see screencast below)

image

### CODE REVIEW

**Comment by eseliger** [comment]
@christinaforney is a changelog entry required for polishing some nits and the simpler insertion? 
If so, do you have some words I can add to the changelog that sound good? :)

**Comment by mrnugget** [comment]
I think the behavior is great, but only if there's a single input field and the "extracting" is hidden from the user. But if you make it a single input field to accept URLs to PRs things get a little tricky, since the repos don't necessarily have to have the same name on Sourcegraph as they do on `$code_host`.

So, I'd keep things simple for now.

I don't like the "name of the repository on Sourcegraph" with the URL text that much.

**Comment by christinaforney** [comment]
Rather than using headings for the field tiles can we use a proper input label? See how we've done this in the profile: 
image

Update hint text to be (@mrnugget what do you think of this):
"Name of repository name in Sourcegraph e.g., codehost/organization/repository"

CHANGELOG entry: "Minor improvements to manual campaign form fields"

**Comment by mrnugget** [comment]
Updated hint text sounds better!

**Comment by felixfbecker** [comment]
> "Name of repository name in Sourcegraph e.g., codehost/organization/repository"

I find this less clear. If `repositoryPathPattern` is configured, then this could be completely wrong. I would imagine the `codehost/` prefix is quite common to be configured away (since companies rarely use multiple code hosts)

File: web/src/enterprise/campaigns/detail/AddChangesetForm.tsx [hunk 1]
```diff
@@ -88,33 +88,80 @@ export const AddChangesetForm: React.FunctionComponent<{ campaignID: ID; onAdd:
...
+                        </p>
+                    </div>
+                    <div className="form-group mr-3 mb-0">
+                        <label htmlFor="changeset-number">
+                            <h3>Changeset number</h3>
+                        </label>
+                        <input
+                            required={true}
+                            name="changeset-number"
+                            type="number"
```
[the remark this answers is not in the document]
Comment (eseliger) [reply web/src/enterprise/campaigns/detail/AddChangesetForm.tsx]: I checked that before and for bbs and github they are both numeric.

File: web/src/enterprise/campaigns/detail/changesets/ChangesetNode.tsx [hunk 2]
```diff
@@ -165,7 +165,9 @@ export const ChangesetNode: React.FunctionComponent<ChangesetNodeProps> = ({
                     ))}
                 </Collapsible>
             ) : (
-                <div className="campaign-node__content flex-fill">{changesetNodeRow}</div>
+                <div className="changeset-node__content changeset-node__content--no-collapse flex-fill">
```
Comment (felixfbecker) [inline web/src/enterprise/campaigns/detail/changesets/ChangesetNode.tsx]: Why does this say `changeset-node__content` but above it says `campaign-node__content` ?

File: web/src/SourcegraphWebApp.scss [hunk 3]
```diff
@@ -93,6 +93,9 @@ $table-border-color-light: #e4e9f1;
 // Progress
 $progress-height: auto;
 
+// Collapsible
+$collapsible-expand-btn-width: 20px;
```
Comment (felixfbecker) [inline web/src/SourcegraphWebApp.scss]: If I am not mistaken.
```suggestion
$collapsible-expand-btn-width: 1.25rem;
```

File: web/src/enterprise/campaigns/detail/AddChangesetForm.tsx [hunk 4]
```diff
@@ -88,33 +88,80 @@ export const AddChangesetForm: React.FunctionComponent<{ campaignID: ID; onAdd:
...
+                            required={true}
+                            name="changeset-repo"
+                            type="text"
+                            size={35}
+                            className="form-control mr-1"
+                            placeholder="Repository name"
+                            value={repoName}
+                            onChange={event => onChangeRepoName(event.target.value)}
+                        />
+                        <p className="form-text text-muted">
```
[the remark this answers is not in the document]
Comment (eseliger) [reply web/src/enterprise/campaigns/detail/AddChangesetForm.tsx]: As per bootstraps form info text, this is not supposed to be indented. Changing it here (changing the `.form-text` class) should probably be a global decision which I would like to not introduce last minute into a patch-release :)

[the remark this answers is not in the document]
Comment (felixfbecker) [reply web/src/enterprise/campaigns/detail/AddChangesetForm.tsx]: kk

File: web/src/enterprise/campaigns/detail/changesets/ChangesetNode.tsx (hunk 2 above)
Comment (eseliger) [reply web/src/enterprise/campaigns/detail/changesets/ChangesetNode.tsx]: that was a typo but they had the same effect 👀

File: web/src/SourcegraphWebApp.scss (hunk 3 above)
Comment (eseliger) [reply web/src/SourcegraphWebApp.scss]: I'll try

Comment (eseliger) [reply web/src/SourcegraphWebApp.scss]: works!

File: web/src/enterprise/campaigns/detail/changesets/ChangesetNode.tsx (hunk 2 above)
Comment (felixfbecker) [reply web/src/enterprise/campaigns/detail/changesets/ChangesetNode.tsx]: Should it be updated above too then?

Comment (eseliger) [reply web/src/enterprise/campaigns/detail/changesets/ChangesetNode.tsx]: 👁

Comment (felixfbecker) [reply web/src/enterprise/campaigns/detail/changesets/ChangesetNode.tsx]: Updated in GitHub editor (untested)

Comment (eseliger) [reply web/src/enterprise/campaigns/detail/changesets/ChangesetNode.tsx]: updated at the same time, works

File: web/src/enterprise/campaigns/detail/AddChangesetForm.tsx [hunk 5]
```diff
@@ -90,31 +90,48 @@ export const AddChangesetForm: React.FunctionComponent<{ campaignID: ID; onAdd:
...
+                            Find the Sourcegraph repository name in the URL (e.g., {window.location.protocol}//
+                            {window.location.host}/<strong>&lt;REPOSITORY_NAME&gt;</strong>)
+                        </p>
+                    </div>
+                    <div className="form-group mr-3 mb-0">
+                        <label htmlFor="changeset-number">Changeset number</label>
+                        <input
+                            required={true}
+                            name="changeset-number"
```
Comment (felixfbecker) [inline web/src/enterprise/campaigns/detail/AddChangesetForm.tsx]: Sorry just noticed this, doesn't this need to be `id` according to MDN?
https://developer.mozilla.org/en-US/docs/Web/API/HTMLLabelElement/htmlFor

> A DOMString which contains the ID string of the element which is associated with the control.

```suggestion
                            id="changeset-number"
```

File: web/src/enterprise/campaigns/detail/AddChangesetForm.tsx [hunk 6]
```diff
@@ -90,31 +90,48 @@ export const AddChangesetForm: React.FunctionComponent<{ campaignID: ID; onAdd:
...
-                    onChange={event => setExternalID(event.target.value)}
-                />
+            <h3>Track changeset</h3>
+            <Form onSubmit={submit}>
+                <div className="d-flex">
+                    <div className="form-group mr-3 mb-0">
+                        <label htmlFor="changeset-repo">Repository name</label>
+                        <input
+                            required={true}
+                            name="changeset-repo"
```
Comment (felixfbecker) [inline web/src/enterprise/campaigns/detail/AddChangesetForm.tsx]:
```suggestion
                            id="changeset-repo"
```

Same suggestion as below

Comment (eseliger) [reply web/src/enterprise/campaigns/detail/AddChangesetForm.tsx]: 👍

**Pull request merged**